habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -0,0 +1,519 @@
+import os
+import time
+from collections import deque
+from typing import Dict, List
+
+import numpy as np
+import torch
+
+from habitat import logger
+from habitat.utils.visualizations.utils import observations_to_image
+from habitat_baselines.common.base_trainer import BaseRLTrainer
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.env_utils import construct_envs
+from habitat_baselines.common.environments import NavRLEnv
+from habitat_baselines.common.tensorboard_utils import get_tensorboard_writer
+from habitat_baselines.common.utils import (
+    RolloutStorage,
+    batch_obs,
+    generate_video,
+    poll_checkpoint_folder,
+    update_linear_schedule,
+)
+from habitat_baselines.rl.ppo import PPO, Policy
+
+
+@baseline_registry.register_trainer(name="ppo")
+class PPOTrainer(BaseRLTrainer):
```

Comment:
Since the actually update rule (in this case PPO) is in a separate class, the vast majority of this trainer is really as `ActorCriticTrainer` and more general than just PPO.

Reply:
This definitely makes sense. I can make a separate PR for this structural change.